pyomo/contrib/appsi/base.py

```diff
@@ -885,7 +911,8 @@ def add_block(self, block):
                 for _p in p.values():
                     param_dict[id(_p)] = _p
         self.add_params(list(param_dict.values()))
-        self.add_variables(list(OrderedDict((id(var), var) for var in block.component_data_objects(Var, descend_into=True, sort=False)).values()))
+        if self._only_child_vars:
+            self.add_variables(list(OrderedDict((id(var), var) for var in block.component_data_objects(Var, descend_into=True, sort=False)).values()))
```

Comment:
I thought `sort=False` would be fine now that sets are ordered? They only way it would be non-deterministic is with `SetOf` with an unordered argument?

Reply:
Sets are ordered by default, but you can still explicitly declare unordered sets. Sort can take on multiple values between completely unordered (False) and sorted indices (True). I do not remember what the current default behavior is, but we should update it to be “deterministic” if that is not currently the case.

Reply:
Thanks, @jsiirola. That is helpful. I made this change.